src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -234,6 +245,32 @@ func (s *commitLogSource) Read(
 				return bootstrap.NamespaceResults{}, err
 			}
 		}
+
+		// Read index snapshot files
+		indexSnapshotFiles, err := s.indexSnapshotFilesFn(
+			filePathPrefix,
+			ns.Metadata.ID(),
+		)
+		if err != nil {
+			return bootstrap.NamespaceResults{}, err
+		}
+		indexInfoFiles := s.readIndexInfoFilesFn(filePathPrefix, ns.Metadata.ID(),
+			fsOpts.InfoReaderBufferSize(), persist.FileSetSnapshotType)
+		// Get latest index snapshot per block start
+		mostRecentIndexSnapshotsByBlock, err := s.mostRecentIndexSnapshotsByBlock(
+			ns.Metadata, shardTimeRanges, indexSnapshotFiles, indexInfoFiles)
+		if err != nil {
+			return bootstrap.NamespaceResults{}, err
+		}
+		indexResult := result.NewIndexBootstrapResult()
+		if err := s.bootstrapIndexSnapshots(
+			ns.Metadata,
+			indexResult,
+			mostRecentIndexSnapshotsByBlock,
+		); err != nil {
+			return bootstrap.NamespaceResults{}, err
+		}
+		indexResults = append(indexResults, indexResult)
```

Comment:
Instead of using a slice here, can you use a map? It's not going to be expensive to use a `map[string]index.Result` since it's just one map per run. 

I'm concerned that the iteration order may change between two iterations of the namespace map. 

`namespaces.Namespaces.Iter()` returns a `map[...]...` and that order is not guaranteed:
https://blog.golang.org/maps
> When iterating over a map with a range loop, the iteration order is not specified and is not guaranteed to be the same from one iteration to the next. If you require a stable iteration order you must maintain a separate data structure that specifies that order. This example uses a separate sorted slice of keys to print a map[int]string in key order: